src/main/java/apoc/export/cypher/formatter/CypherFormatterUtils.java

```diff
@@ -0,0 +1,185 @@
+package apoc.export.cypher.formatter;
+
+import apoc.export.util.FormatUtils;
+import org.neo4j.graphdb.Label;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.PropertyContainer;
+import org.neo4j.helpers.collection.Iterables;
+
+import java.lang.reflect.Array;
+import java.util.*;
+
+/**
+ * @author AgileLARUS
+ *
+ * @since 16-06-2017
+ */
+public class CypherFormatterUtils {
+
+	public final static String UNIQUE_ID_LABEL = "UNIQUE IMPORT LABEL";
+	public final static String UNIQUE_ID_PROP = "UNIQUE IMPORT ID";
+	public final static String Q_UNIQUE_ID_LABEL = quote(UNIQUE_ID_LABEL);
+
+    // ---- common ----
+
+    public static String quote(String id) {
+        return "`" + id + "`";
+    }
+
+    public static String label(String id) {
+        return ":`" + id + "`";
+    }
+
+    // ---- label ----
+
+	public static String labelsAsString(Node node, Map<String, String> uniqueConstraints, Set<String> indexNames) {
+		Iterator<Label> labels = node.getLabels().iterator();
+		StringBuilder result = new StringBuilder(100);
+		boolean uniqueFound = false;
+		while (labels.hasNext()) {
+			Label next = labels.next();
+			String labelName = next.name();
+			if (uniqueConstraints.containsKey(labelName) && node.hasProperty(uniqueConstraints.get(labelName)))
+				uniqueFound = true;
+			if (indexNames.contains(labelName))
+				result.insert(0, label(labelName));
+			else
+				result.append(label(labelName));
+		}
+		if (!uniqueFound) {
+			result.append(label(UNIQUE_ID_LABEL));
+		}
+		return result.toString();
+	}
+
+    public static String notUniqueLabelsAsString(String id, Node node, Map<String, String> uniqueConstraints) {
+
+        String nodeLookupLabel = null;
```

Comment:
perhaps we should even have a method that takes node-labels and returns a map with label -> "indexed/unique" that we can iterate/filter over